apps/web/components/settings/ChangeEmailAddress.tsx

```diff
@@ -0,0 +1,122 @@
+"use client";
+
+import type { z } from "zod";
+import { ActionButton } from "@/components/ui/action-button";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { toast } from "@/components/ui/use-toast";
+import { api } from "@/lib/trpc";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { signOut } from "next-auth/react";
+import { useForm } from "react-hook-form";
+
+import { zChangeEmailAddressSchema } from "@hoarder/shared/types/users";
+
+export function ChangeEmailAddress() {
+  const form = useForm<z.infer<typeof zChangeEmailAddressSchema>>({
+    resolver: zodResolver(zChangeEmailAddressSchema),
+    defaultValues: {
+      newEmailAddress: "",
+      currentPassword: "",
+    },
+  });
+
+  const mutator = api.users.changeEmailAddress.useMutation({
+    onSuccess: () => {
+      toast({
+        description: "Email Address changed successfully. Logging out.",
+      });
+      form.reset();
+      signOut();
```

Comment:
hmmm, this gets a bit tricky with JWTs. Yes here you'll sign out this session, but all other sessions will continue having a valid JWT which can get problematic. I really hate that next auth is forcing us to use JWTs :(